nailgun/entities.py

```diff
@@ -423,74 +422,96 @@ def __init__(self, server_config=None, **kwargs):
-                required=True,
             ),
-            'region': entity_fields.StringField(null=True),
-            'server': entity_fields.StringField(null=True),
-            'set_console_password': entity_fields.BooleanField(null=True),
-            'tenant': entity_fields.StringField(null=True),
+            'provider_friendly_name': entity_fields.StringField(),
             'url': entity_fields.URLField(required=True),
-            'user': entity_fields.StringField(null=True),
-            'uuid': entity_fields.StringField(null=True),
         }
         self._meta = {
             'api_path': 'api/v2/compute_resources',
             'server_modes': ('sat'),
         }
-        super(ComputeResource, self).__init__(server_config, **kwargs)
+        super(AbstractComputeResource, self).__init__(server_config, **kwargs)
 
-    def create_missing(self):
-        """Customize the process of auto-generating instance attributes.
+    def create_payload(self):
+        """Wrap submitted data within an extra dict.
+
+        For more information, see `Bugzilla #1151220
+        <https://bugzilla.redhat.com/show_bug.cgi?id=1151220>`_.
+
+        """
+        return {
+            u'compute_resource': super(
+                AbstractComputeResource,
+                self
+            ).create_payload()
+        }
+
+
+class DockerComputeResource(AbstractComputeResource):
+    """A representation of a Docker Compute Resource entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        super(DockerComputeResource, self).__init__(server_config, **kwargs)
+        self._fields.update({
+            'email': entity_fields.EmailField(),
+            'password': entity_fields.StringField(null=True),
+            'url': entity_fields.URLField(required=True),
+            'user': entity_fields.StringField(null=True),
+        })
+        self._fields['provider'].default = 'Docker'
+        self._fields['provider'].required = True
+        self._fields['provider_friendly_name'].default = 'Docker'
+
+    def create(self, create_missing=True):
+        """Do extra work to fetch a complete set of attributes for this entity.
+
+        For more information, see `Bugzilla #1223540
+        <https://bugzilla.redhat.com/show_bug.cgi?id=1223540>`_.
+
+        """
+        return DockerComputeResource(
+            self._server_config,
+            id=self.create_json(create_missing)['id'],
+        ).read()
+
+    def read(self, entity=None, attrs=None, ignore=('password',)):
+        """Do extra work to fetch a complete set of attributes for this entity.
+
+        For more information, see `Bugzilla #1223540
+        <https://bugzilla.redhat.com/show_bug.cgi?id=1223540>`_.
+
+        Also, do not try to read the "password" field. No value is returned for
+        the field, for obvious reasons.
+
+        """
+        if attrs is None:
+            attrs = self.read_json()
+        if 'email' not in attrs and 'email' not in ignore:
```

Comment:
When an HTTP GET request is issued to `/api/v2/compute_resources` for a docker compute resource, an "email" attribute should be returned. None is returned. However, the "email" attribute is returned in response to `PUT` requests. So we make `PUT` request here, grab the email, and insert the email into `attrs`.